app/views/AssessmentRegistryForm/MetadataForm/StakeholderForm/index.tsx

```diff
@@ -0,0 +1,162 @@
+}
+
+function StakeholderForm(props: Props) {
+    const {
+        className,
+        value,
+        setValue,
+        loading,
+        error: riskyError,
+    } = props;
+    const [stakeholderOptions, setStakeholderOptions] = useState<BasicOrganization[]>([]);
+
+    const error = getErrorObject(riskyError);
+    const errorMap = useMemo(
+        () => ({
+            LEAD_ORGANIZATION: getErrorString(error?.leadOrganizations),
+            DONOR: getErrorString(error?.donors),
+            INTERNATIONAL_PARTNER: getErrorString(error?.internationalPartners),
+            NATIONAL_PARTNER: getErrorString(error?.nationalPartners),
+            GOVERNMENT: getErrorString(error?.governments),
+        }),
+        [error],
+    );
+
+    const organizationListRendererParams = useCallback(
+        (key: ProjectOrganizationTypeEnum, v: StakeholderType) => ({
+            data: stakeholderOptions.filter((org) => value[v.formId]?.includes(org.id)),
+            title: v.label,
+            dataPending: loading,
+            error: errorMap[key],
+        }),
+        [loading, errorMap, value, stakeholderOptions],
+    );
+
+    const handleOrganizationChange = useCallback(
+        (values: BasicProjectOrganization[] | undefined) => {
+            const groupedStakeholders = listToGroupList(
+                (values ?? []).filter((org) => org.organizationType),
+                (o) => o.organizationType,
+                (o) => o.organization,
+            );
+
+            setValue({
+                ...value,
+                donors: groupedStakeholders.DONOR ?? undefined,
+                leadOrganizations: groupedStakeholders.LEAD_ORGANIZATION ?? undefined,
+                nationalPartners: groupedStakeholders.NATIONAL_PARTNER ?? undefined,
+                internationalPartners: groupedStakeholders.INTERNATIONAL_PARTNER ?? undefined,
+                governments: groupedStakeholders.GOVERNMENT ?? undefined,
+            });
+        }, [value, setValue],
+    );
+
+    const organizations = useMemo(
+        () => {
+            const stakeholderValueOnly = {
+                leadOrganizations: value.leadOrganizations,
+                internationalPartners: value.internationalPartners,
+                nationalPartners: value.nationalPartners,
+                donors: value.donors,
+                governments: value.governments,
+            };
+            return Object.entries(stakeholderValueOnly).flatMap(([key, val]) => {
+                if (val) {
+                    const matchedOrganizationData = stakeholderTypes.find((d) => d.formId === key);
+                    if (matchedOrganizationData) {
+                        return val.map((organization) => ({
+                            organization,
+                            organizationType: matchedOrganizationData.id,
+                        }));
+                    }
+                }
+                return [];
+            });
+        },
+        [value],
+    );
```

Comment:
@puranban , for each item just use a map, this is complicated. If the no of stakeholder types were more than 10 this might make sense.